crates/uroborosql-fmt-wasm/src/lib.rs

```diff
@@ -1,7 +1,30 @@
 use std::ffi::{c_char, CStr, CString};
 
+static mut RESULT: &mut [u8] = &mut [0; 50000];
+static mut ERROR_MSG: &mut [u8] = &mut [0; 50000];
```

Comment:
エラーメッセージはともかく、フォーマット結果の格納先が50KBは少ないかもしれません。
固定長でいくつまで持てばいいと決めるのは難しいので、CStringで持っておくのはどうでしょうか。
以下手元で動かしてみた時のpatchです。ご参考までに。

<details>
<summary>手元で動かしたときのpatch</summary>

```patch
diff --git a/crates/uroborosql-fmt-wasm/Cargo.toml b/crates/uroborosql-fmt-wasm/Cargo.toml
index 23df84b..8335e7f 100644
--- a/crates/uroborosql-fmt-wasm/Cargo.toml
+++ b/crates/uroborosql-fmt-wasm/Cargo.toml
@@ -13,4 +13,5 @@ repository.workspace = true
 crate-type = ["cdylib"]
 
 [dependencies]
+once_cell = "1.17.0"
 uroborosql-fmt = { workspace = true }
diff --git a/crates/uroborosql-fmt-wasm/src/lib.rs b/crates/uroborosql-fmt-wasm/src/lib.rs
index 38f47fa..71fa7d5 100644
--- a/crates/uroborosql-fmt-wasm/src/lib.rs
+++ b/crates/uroborosql-fmt-wasm/src/lib.rs
@@ -1,18 +1,22 @@
-use std::ffi::{c_char, CStr, CString};
-
-static mut RESULT: &mut [u8] = &mut [0; 50000];
-static mut ERROR_MSG: &mut [u8] = &mut [0; 50000];
+use std::{
+    ffi::{c_char, CStr, CString},
+    sync::Mutex,
+};
 
+use once_cell::sync::Lazy;
 use uroborosql_fmt::{config::Config, format_sql_with_config};
 
+static RESULT: Lazy<Mutex<CString>> = Lazy::new(|| Mutex::new(CString::new("").unwrap()));
+static ERROR_MSG: Lazy<Mutex<CString>> = Lazy::new(|| Mutex::new(CString::new("").unwrap()));
+
 /// Returns the address of the result string.
 ///
 /// # Safety
 ///
 /// This is unsafe because it returns a raw pointer.
 #[no_mangle]
-pub unsafe extern "C" fn get_result_address() -> *const u8 {
-    &RESULT[0]
+pub unsafe extern "C" fn get_result_address() -> *const c_char {
+    RESULT.lock().unwrap().as_c_str().as_ptr()
 }
 
 /// Returns the address of the error message string.
@@ -21,8 +25,8 @@ pub unsafe extern "C" fn get_result_address() -> *const u8 {
 ///
 /// This is unsafe because it returns a raw pointer.
 #[no_mangle]
-pub unsafe extern "C" fn get_error_msg_address() -> *const u8 {
-    &ERROR_MSG[0]
+pub unsafe extern "C" fn get_error_msg_address() -> *const c_char {
+    ERROR_MSG.lock().unwrap().as_c_str().as_ptr()
 }
 
 /// Formats SQL code given as char pointer `src` by WASM (JavaScript).
@@ -33,52 +37,20 @@ pub unsafe extern "C" fn get_error_msg_address() -> *const u8 {
 /// [`CStr::from_ptr`](https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.from_ptr).
 #[export_name = "format_sql"]
 #[no_mangle]
-pub unsafe extern "C" fn format_sql_for_wasm(src: *mut c_char, config_json_str: *mut c_char) {
+pub unsafe extern "C" fn format_sql_for_wasm(src: *const c_char, config_json_str: *const c_char) {
+    // Clear previous format result
+    *RESULT.lock().unwrap() = CString::new("").unwrap();
+    *ERROR_MSG.lock().unwrap() = CString::new("").unwrap();
+
     let src = CStr::from_ptr(src).to_str().unwrap().to_owned();
 
-    let config_json_str = CStr::from_ptr(config_json_str).to_str().unwrap();
+    let config_json_str: &str = CStr::from_ptr(config_json_str).to_str().unwrap();
     let config = Config::from_json_str(config_json_str).unwrap();
 
     let result = format_sql_with_config(&src, config);
 
     match result {
-        Ok(result) => {
-            CString::new(result)
-                .unwrap()
-                .as_bytes_with_nul()
-                .iter()
-                .enumerate()
-                .for_each(|(i, x)| {
-                    RESULT[i] = *x;
-                });
-
-            CString::new("")
-                .unwrap()
-                .as_bytes_with_nul()
-                .iter()
-                .enumerate()
-                .for_each(|(i, x)| {
-                    ERROR_MSG[i] = *x;
-                });
-        }
-        Err(err) => {
-            CString::new(err.to_string())
-                .unwrap()
-                .as_bytes_with_nul()
-                .iter()
-                .enumerate()
-                .for_each(|(i, x)| {
-                    ERROR_MSG[i] = *x;
-                });
-
-            CString::new("")
-                .unwrap()
-                .as_bytes_with_nul()
-                .iter()
-                .enumerate()
-                .for_each(|(i, x)| {
-                    RESULT[i] = *x;
-                });
-        }
+        Ok(result) => *RESULT.lock().unwrap() = CString::new(result).unwrap(),
+        Err(err) => *ERROR_MSG.lock().unwrap() = CString::new(err.to_string()).unwrap(),
     }
 }
diff --git a/wasm/main.js b/wasm/main.js
index db15f09..1e2baf6 100644
--- a/wasm/main.js
+++ b/wasm/main.js
@@ -133,10 +133,6 @@ clear_all_button.addEventListener("click", () => clear_all_config());
 
 // wasmの初期化が終了した際の処理
 function initialize() {
-  // Rust側で確保したメモリのポインタを取得
-  const result_ptr = ccall("get_result_address", "number", [], []);
-  const error_ptr = ccall("get_error_msg_address", "number", [], []);
-
   function formatSql() {
     if (!src_editor || !dst_editor) {
       console.log("editors have not been loaded.");
@@ -237,6 +233,10 @@ function initialize() {
     // 何ミリ秒かかったかを表示する
     console.log("format complete: " + (endTime - startTime) + "ms");
 
+    // Rust側で確保したメモリのポインタを取得
+    const result_ptr = ccall("get_result_address", "number", [], []);
+    const error_ptr = ccall("get_error_msg_address", "number", [], []);
+
     // Module.UTF8ToString() でポインタを js の string に変換
     const res = UTF8ToString(result_ptr);
     const err = UTF8ToString(error_ptr);
```
</details>

Reply:
ご指摘ありがとうございます。
フォーマット結果、エラー共にCStringで持つように変更しました。
変更前は100000行ほどでフォーマットエラーが発生していたのですが、この変更により解消されました。